peps/pep-9999.rst

```diff
@@ -0,0 +1,546 @@
+        # error: MySubclass does not override "foo"
+
+        def pprint(self) -> None:
+            print(self.foo, self.bar, self.baz)
+
+* In a protocol attribute declaration, ``name: ReadOnly[T]`` indicates that a structural
+  subtype must support ``.name`` access, and the returned value is compatible with ``T``::
+
+    class HasName(Protocol):
+        name: ReadOnly[str]
+
+
+    class NamedAttr:
+        name: str
+
+    class NamedProp:
+        @property
+        def name(self) -> str: ...
+
+    class NamedClassVar:
+        name: ClassVar[str]
+
+    class NamedDescriptor:
+        @cached_property
+        def name(self) -> str: ...
+
+    # all of the following are ok
+    has_name: HasName
+    has_name = NamedAttr()
+    has_name = NamedProp()
+    has_name = NamedClassVar
+    has_name = NamedClassVar()
+    has_name = NamedDescriptor()
+
+Interaction with other special types
+------------------------------------
+
+``ReadOnly`` can be used with ``ClassVar`` and ``Annotated`` in any nesting order:
+
+.. code-block:: python
+
+    class Foo:
+        foo: ClassVar[ReadOnly[str]] = "foo"
+        bar: Annotated[ReadOnly[int], Gt(0)]
+
+.. code-block:: python
+
+    class Foo:
+        foo: ReadOnly[ClassVar[str]] = "foo"
+        bar: ReadOnly[Annotated[int, Gt(0)]]
+
+This is consistent with the interaction of ``ReadOnly`` and :class:`typing.TypedDict`
+defined in :pep:`705`.
+
+``ClassVar`` excludes read-only attributes from being assignable to within
+initialization methods.
+
+Rules of ``Final`` should take priority when combined with ``ReadOnly``. As such,
+type checkers may warn on the redundancy of combining the two type qualifiers.
+
+
+Backwards Compatibility
+=======================
+
+This PEP introduces new contexts where ``ReadOnly`` is valid. Programs inspecting
+those places will have to change to support it. This is expected to mainly affect type checkers.
+
+However, caution is advised while using the backported ``typing_extensions.ReadOnly``
+in older versions of Python. Mechanisms inspecting annotations may behave incorrectly
+when encountering ``ReadOnly``; in particular, the ``@dataclass`` decorator
+which `looks for <https://docs.python.org/3/library/dataclasses.html#class-variables>`_
+``ClassVar`` will incorrectly treat ``ReadOnly[ClassVar[...]]`` as an instance attribute.
+
+
+Security Implications
+=====================
+
+There are no known security consequences arising from this PEP.
+
+
+How to Teach This
+=================
+
+[How to teach users, new and experienced, how to apply the PEP to their work.]
+
+
+Open Issues
+===========
+
+Assignment in ``__new__``
+-------------------------
+
+Immutable classes like :class:`fractions.Fraction` often do not define ``__init__``;
+instead, they perform initialization in ``__new__`` or classmethods. The proposed
+feature won't be useful to them. 
+
+OTOH, allowing assignment within ``__new__`` (and/or classmethods) could open way
```

Comment:
Expand this abbreviation

Reply:
I have assumed I could be less formal in the `Open Issues` section, my bad 😄